File: app/src/main/java/com/bakkenbaeck/sol/service/BootReceiver.java
package com.bakkenbaeck.sol.service;


import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class BootReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(final Context context, final Intent intent) {
        if (intent.getAction().equals("android.intent.action.BOOT_COMPLETED")) {
            final Intent service = new Intent(context, SunsetService.class);
            service.putExtra(SunsetService.EXTRA_SHOW_NOTIFICATION, false);
            context.startService(service);
        }
    }
}

File: app/src/main/java/com/bakkenbaeck/sol/service/BootReceiver.kt
package com.bakkenbaeck.sol.service


import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.bakkenbaeck.sol.BaseApplication

class BootReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action == "android.intent.action.BOOT_COMPLETED") {
            BaseApplication.instance.refreshLocation(false)
        }
    }
}